inst/report/scripts/run_molevolvr_pipeline.R

```diff
@@ -0,0 +1,1296 @@
+                     "Pfam", "SMART", "Phobius",
+                     "Gene3D", "TMHMM", "SignalP_GRAM_POSITIVE",
+                     "SUPERFAMILY", "MobiDBLite", "TIGRFAM", "PANTHER", "Coils"
+                   )) {
+  # read in cleaned up iprscan results
+  ipr_in <- read_tsv(infile_ipr, col_names = T, col_types = ipr_cln_cols)
+
+  # split dataframe into unique proteins
+  x <- split(x = ipr_in, f = ipr_in$AccNum)
+
+  # plan(strategy = "multicore", .skip = T)
+
+  # within each data.table
+  domarch <- map(x, function(y) {
+    # domarch <- future_map(x, function(y) {
+    acc_row <- data.frame(AccNum = y$AccNum[1], stringsAsFactors = F)
+    DAs <- data.frame(matrix(nrow = 1, ncol = length(analysis)))
+    DA <- y %>%
+      group_by(Analysis) %>%
+      arrange(StartLoc)
+    i <- 1
+    for (a in analysis) {
+      a_da <- DA %>% filter(Analysis == a)
+      if (a == "SignalP_EUK" || a == "SignalP_GRAM_NEGATIVE" ||
+          a == "SignalP_GRAM_POSITIVE") {
+        var_shortname <- "DB.ID"
+      } else {
+        var_shortname <- "ShortName"
+      }
+      var_shortname_sym <- sym(var_shortname)
+      a_da <- a_da %>%
+        ungroup() %>%
+        select({{ var_shortname_sym }}) %>%
+        filter(!is.na({{ var_shortname_sym }})) %>%
+        filter(!is.null({{ var_shortname_sym }})) %>%
+        pull(var_shortname) %>%
+        paste(collapse = "+")
+      DAs[1, i] <- a_da
+      i <- (i + 1)
+    }
+
+    colnames(DAs) <- paste("DomArch", analysis, sep = ".")
+    return(cbind(acc_row, DAs))
+  })
+
+  # select relevant rows from ipr input to add to domarch
+  ipr_select <- ipr_in %>%
+    select(Name, AccNum, Species, TaxID, Lineage, Lineage_long_na,
+           Lineage_long, Lineage_med, Lineage_short, ProteinName,
+           SourceDB, Completeness, AccNum.noV) %>%
+    distinct()
+
+  # combine domarchs to one data frame, merge w/ acc2info
+  domarch2 <- do.call(rbind.data.frame, domarch)
+
+  domarch_lins <- domarch2 %>%
+    merge(ipr_select, by = "AccNum", all.x = T)
+
+  # save domarch_lins file
+  write_tsv(domarch_lins,
+            file = paste0(prefix, "/", prefix, ".ipr_domarch.tsv"),
+            append = F, na = "NA"
+  )
+
+  # return domarch2 dataframe to append to blast results if given
+  return(domarch2)
+}
+
+## function to add results from ipr2da to blast results
+append_ipr <- function(ipr_da, blast, prefix) {
+  # ! an 'AccNum' or 'AccNum.noV' column is required in blast table for joining !
+  blast_out <- read_tsv(blast, col_names = T)
+  if ("AccNum.noV" %in% colnames(blast_out)) {
+    ipr_da <- read_tsv(paste0(prefix, "/", prefix, ".ipr_domarch.tsv"), col_names = T)
+    blast_ipr <- merge(blast_out, ipr_da, by = "AccNum.noV", all.x = T)
+  } else {
+    blast_ipr <- merge(blast_out, ipr_da, by = "AccNum", all.x = T)
+  }
+
+  write_tsv(blast_ipr, file = paste0(prefix, "/", prefix, ".full_analysis.tsv"), na = "NA")
+}
+
+# Web BLAST output
+web_blast_colnames <- c("Query", "AccNum",
```

Comment:
L1051–1246: column name rename functions -- this needn't be recalled here, either. 

@epbrenner @AbhirupaGhosh @charmvang, here are the guidelines I mentioned for colnames to be adopted for CARD data, too.